### PR TITLE
feat: Protocol Specification changes 2

### LINKED ISSUE #181: Protocol Spec missing Delete Verb
## What needs to be done

Protocol specification missing the delete verb

This is what you mentioned in stand, right @gkc ?

### LINKED ISSUE #198: Protocol Spec update Notification Verb
## What needs to be done

Update the [Protocol Specification](https://docs.atsign.com/atplatform/specification/) with new notification examples + their responses, and a new regex.

1. Notification regex to be updated with `messageType:` parameter. 

New regex to be added:
```
notify:((?<operation>update|delete):)?(messageType:(?<messageType>key|text):)?(priority:(?<priority>low|medium|high):)?(strategy:(?<strategy>all|latest):)?(latestN:(?<latestN>\d+):)?(notifier:(?<notifier>[^\s:]+):)?(ttln:(?<ttln>\d+):)?(ttl:(?<ttl>\d+):)?(ttb:(?<ttb>\d+):)?(ttr:(?<ttr>(-)?\d+):)?(ccd:(?<ccd>true|false):)?(@(?<forAtSign>[^@:\s]*)):(?<atKey>[^:@]((?!:{2})[^@])+)(@(?<atSign>[^@:\s]+))?(:(?<value>.+))?$
```

2. Add examples of `notify:messageType:text:` and their responses
3. Add the `notify:status:` verb with examples and their responses

## Support

[Discord](https://discord.atsign.com)

### PR DESCRIPTION
**Pull request by JeremyTubongbanua**
## What I did

- Added a table under `# Verbs` categorizing each verb's purpose.
- Refactored the `pkam` verb by moving it with the rest of the authentication verbs
- Added `plookup:bypassCache:true` as one of the examples
- Added the `delete` verb
- Added `messageType:text` as an example in the `notify` verb
- Updated `notify` verb regex
- Added `notify:status` documentation
- prefix `key:` to `data:` under `from` verb response section
- moved `update` and `update:meta` to above the `delete` verb

## TO DO

- [x] plookup automatic cache explanation (plookup'd public keys are cached by default)
- [x] updatedOn metadata typo
- [x] give titles for each verb example (especially notify:Delete which is confusing)
- [x] ttr 0, -1, > 0, and null, explanation differences
- [x] from response ?

## Screenshots

Verb Table
image

The delete verb
image

Notify:status verb
image

Plookup verb
image

closes #181 
closes #198 

This PR is the same as #245 but with resolved merge conflicts

### CODE REVIEW

File: content/docs/atplatform/specification.md
Comment (xavierchanth) [inline content/docs/atplatform/specification.md:206]: Can we move `update` and `update:meta` down so that things are sorted by purpose? And also, does it make sense to do so.

File: content/docs/atplatform/specification.md
Comment (xavierchanth) [inline content/docs/atplatform/specification.md]: I think that sync should have a distinct purpose, maybe "data synchronization".

Comment (JeremyTubongbanua) [reply content/docs/atplatform/specification.md]: done in https://github.com/atsign-foundation/docs.atsign.com/pull/247/commits/61e3d28ef1f5d44f17fd2b4221b9dbf582ba5c53

File: content/docs/atplatform/specification.md
Comment (JeremyTubongbanua) [reply content/docs/atplatform/specification.md:206]: done in https://github.com/atsign-foundation/docs.atsign.com/pull/247/commits/b5f03f48e7a146253a6a1d9d8eb63289cf75304e

**Review by xavierchanth — APPROVED** [review]

**Review by xavierchanth — DISMISSED** [review]

**Comment by JeremyTubongbanua** [comment]
Would like your review @gkc before merge

File: content/docs/atplatform/specification.md
Comment (gkc) [inline content/docs/atplatform/specification.md:149]: Is null treated the same as 0 ?

Comment (gkc) [inline content/docs/atplatform/specification.md]: Is null treated the same as 0 ?

File: content/docs/atplatform/specification.md
Comment (gkc) [inline content/docs/atplatform/specification.md]: 'user' is confusing here.

The info that needs to be clearly conveyed is:
from+cram : Client program connecting to the atServer
from+pkam: Client program connecting to the atServer
from+pol: atServer connecting to another atServer

**Review by gkc — CHANGES_REQUESTED** [review]
This is a really dramatic improvement to this doc, thank you!!!

I need to read it at more length again but have a couple of requests to start with

File: content/docs/atplatform/specification.md
Comment (JeremyTubongbanua) [reply content/docs/atplatform/specification.md]: changes made here:

- https://github.com/atsign-foundation/docs.atsign.com/pull/247/commits/5ad166d0c1af51e1d17aba72db260528bea554d2
- https://github.com/atsign-foundation/docs.atsign.com/pull/247/commits/f2a534e8b0fd215e0332fedbf2197d308e681272

File: content/docs/atplatform/specification.md
Comment (JeremyTubongbanua) [reply content/docs/atplatform/specification.md]: i mentioned they have the same effect (do not cache) in this commit:
https://github.com/atsign-foundation/docs.atsign.com/pull/247/commits/eba0d8250569096d5434e7b208b366ed9ef4e590

File: content/docs/atplatform/specification.md
Comment (gkc) [inline content/docs/atplatform/specification.md]: "indicates another" should be "indicates to another"

**Review by gkc — CHANGES_REQUESTED** [review]

**Comment by gkc** [comment]
Looks really good @JeremyTubongbanua ! Spotted one more typo you need to fix (https://github.com/atsign-foundation/docs.atsign.com/pull/247#discussion_r1067932932) ... but I'll approve the PR now so you're not waiting for me to approve again after you fix it.

**Review by gkc — DISMISSED** [review]
Looks really good @JeremyTubongbanua ! Spotted one more typo you need to fix (https://github.com/atsign-foundation/docs.atsign.com/pull/247#discussion_r1067932932) ... but I'll approve the PR now so you're not waiting for me to approve again after you fix it.

File: content/docs/atplatform/specification.md
Comment (JeremyTubongbanua) [reply content/docs/atplatform/specification.md]: https://github.com/atsign-foundation/docs.atsign.com/pull/247/commits/9482502c4ed250b596d3396fff29b604cb30e7af

**Comment by JeremyTubongbanua** [comment]
Oops,, we have the GitHub setting enabled where it asks for a review again if I push code.

@XavierChanth can you please approve and merge if you think it's good?

**Review by gkc — APPROVED** [review]

**Comment by JeremyTubongbanua** [comment]
@XavierChanth Please merge

**Comment by cpswan** [comment]
Merged, as I found myself here anyway.

**Pull request merged**
